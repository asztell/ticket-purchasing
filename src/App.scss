.App-Content {
  //   width: 1000px;
  width: 80vw;
  //   border: 1px solid red;
  margin-left: auto;
  margin-right: auto;
}

.App-Header {
  /* background-color: #282c34; */
  background-color: rgb(2, 108, 223);
  height: 60px;
  /* min-height: 100vh; */
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  font-size: calc(10px + 2vmin);
  margin-bottom: 50px;
  /* color: white; */
}
