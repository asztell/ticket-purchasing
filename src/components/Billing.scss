.Delivery,
.Payment,
.Ticket-Insurance {
  border: 1px solid rgb(191, 191, 191);
  border-radius: 4px;
  padding-left: 20px;
  padding-right: 20px;
  margin-bottom: 20px;
  & > h2 {
    font-size: 20px;
    font-family: "Averta", "Helvetica Neue", Helvetica, Arial, sans-serif;
  }
}

.Delivery {
  & h3 {
    font-size: 17px;
    font-family: "Averta", "Helvetica Neue", Helvetica, Arial, sans-serif;
    font-weight: 500;
  }
}

.Payment-Section {
  display: flex;
  flex-direction: column;
  margin-bottom: 20px;
  & > input[type="text"] {
    background-color: rgb(255, 255, 255);
    border-radius: 2px;
    border: 1px solid rgb(153, 153, 153);
    box-shadow: rgba(0, 0, 0, 0.08) 0px 2px 2px 0px inset;
    font-size: 16px;
    line-height: 35px;
    margin-top: 4px;
    outline: none;
    padding: 4px 32px 4px 16px;
  }
  & > input[type="text"]:focus {
    border-color: rgb(0, 132, 255);
    box-shadow: rgba(0, 132, 255, 0.3) 0px 2px 8px 0px inset;
  }
  & > input[type="month"] {
    margin-top: 4px;
    font-family: "Averta", "Helvetica Neue", Helvetica, Arial, sans-serif;
    font-size: 16px;
  }
}
.Payment {
  .Payment-Validation-Error {
    color: rgb(217, 58, 58);
    font-size: 12px;
    line-height: 15px;
    margin-top: 4px;
    word-break: break-word;
    transition: opacity 300ms cubic-bezier(0.455, 0.03, 0.515, 0.955) 0s,
      margin-bottom 300ms cubic-bezier(0.455, 0.03, 0.515, 0.955) 0s;
    opacity: 1;
    margin-bottom: 0px;
  }

  /* shamelessly stolen code from ticketmaster for cuteness purposes */
  .Amex {
    display: inline-block;
    border-radius: 2px;
    margin-top: 4px;
    background: url(https://checkout.ticketmaster.com/fanwallet/v4.66.1/static/images/wallet-items/american-express.svg)
      center center / 32px 24px no-repeat transparent;
    width: 32px;
    min-width: 32px;
    height: 24px;
  }
  .Visa {
    display: inline-block;
    border-radius: 2px;
    margin-top: 4px;
    background: url(https://checkout.ticketmaster.com/fanwallet/v4.66.1/static/images/wallet-items/visa.svg)
      center center / 32px 24px no-repeat transparent;
    width: 32px;
    min-width: 32px;
    height: 24px;
  }
  .Master-Card {
    display: inline-block;
    border-radius: 2px;
    margin-top: 4px;
    background: url(https://checkout.ticketmaster.com/fanwallet/v4.66.1/static/images/wallet-items/master-card.svg)
      center center / 32px 24px no-repeat transparent;
    width: 32px;
    min-width: 32px;
    height: 24px;
  }
  .Discover {
    display: inline-block;
    border-radius: 2px;
    margin-top: 4px;
    background: url(https://checkout.ticketmaster.com/fanwallet/v4.66.1/static/images/wallet-items/discover.svg)
      center center / 32px 24px no-repeat transparent;
    width: 32px;
    min-width: 32px;
    height: 24px;
  }
  .Diners-Club {
    display: inline-block;
    border-radius: 2px;
    margin-top: 4px;
    background: url(https://checkout.ticketmaster.com/fanwallet/v4.66.1/static/images/wallet-items/diners-club.svg)
      center center / 32px 24px no-repeat transparent;
    width: 32px;
    min-width: 32px;
    height: 24px;
  }
  .Accepted-Cards {
    display: flex;
  }
  .Accepted-Cards > * {
    margin-right: 8px;
    margin-bottom: 25px;
  }
}
