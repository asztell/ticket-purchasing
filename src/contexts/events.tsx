import { createContext, useEffect, useMemo, useState } from "react";

export type Event = {
  id: string;
  name: string;
  location: string;
  ISODate: string;
  price: number;
};

export type EventsContextType = {
  events: Event[];
  error: string | null;
};

export const EventsContext = createContext<EventsContextType>(
  {} as EventsContextType
);

export function EventsProvider({ children }: { children: React.ReactNode }) {
  const [events, setEvents] = useState<Event[]>([]);
  const [error, setError] = useState<string | null>(null);

  useEffect(() => {
    fetch(`http://localhost:8080/events`)
      .then((res) => {
        if (!res.ok) {
          throw res;
        }
        return res.json();
      })
      .then((result) => {
        setEvents(result);
      })
      .catch((error) => {
        setError(error);
      });
  }, [setEvents]);

  const value = useMemo(() => {
    return {
      events,
      error,
    };
  }, [events, error]);

  return (
    <EventsContext.Provider value={value}>{children}</EventsContext.Provider>
  );
}
