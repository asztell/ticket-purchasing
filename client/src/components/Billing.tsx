import { useState, useContext, useCallback } from 'react'
import { FormattedMessage } from 'react-intl'
import { TicketPurchasingContext } from '../contexts'
import './Billing.scss'

export function Billing({ className }: { className?: string }): JSX.Element {
  const [showSecurityCodeValidation, setShowSecurityCodeValidation] =
    useState(false)
  const [showCardNumberValidation, setShowCardNumberValidation] =
    useState(false)

  const { cardInfo, updateCardInfo } = useContext(TicketPurchasingContext)
  const [showExpirationDateValidation, setShowExpirationDateValidation] =
    useState(false)

  const handlePaymentChanges = useCallback(
    (event: React.ChangeEvent<HTMLInputElement>) => {
      if (cardInfo !== null) {
        updateCardInfo({
          ...cardInfo,
          [event.target.className]: event.target.value
        })
      }
    },
    [cardInfo, updateCardInfo]
  )

  const handleCardNumberBlur = useCallback(() => {
    setShowCardNumberValidation(true)
  }, [setShowCardNumberValidation])

  function displayValidation(fieldName: string): string | null {
    switch (fieldName) {
      case 'cardNumber':
        if (showCardNumberValidation) {
          return cardInfo?.cardType !== null ? null : ''
        } else {
          return null
        }
      case 'securityCode':
        if (showSecurityCodeValidation) {
          return cardInfo?.securityCodeValid !== null ? null : 'Invalid'
        } else {
          return null
        }
      case 'expirationDate':
        if (showExpirationDateValidation) {
          return cardInfo?.expirationDateValid !== null ? null : 'Invalid'
        } else {
          return null
        }
      default:
        return null
    }
  }

  const handleSecurityCodeBlur = useCallback(() => {
    setShowSecurityCodeValidation(true)
  }, [setShowSecurityCodeValidation])

  const handleExpirationDateBlur = useCallback(() => {
    setShowExpirationDateValidation(true)
  }, [setShowExpirationDateValidation])

  return (
    <div className={className}>
      {/* TODO: each Billing entry can be extracted into its own component */}
      <div className='Delivery'>
        <h2 className='Section-Header'>
          <FormattedMessage id='Billing.Delivery.Header.Text' />
        </h2>
        <h3>
          <FormattedMessage id='Billing.Delivery.Sub-Header.Text' />
        </h3>
        <p>
          <FormattedMessage id='Billing.Delivery.Description.Text' />
        </p>
      </div>
      <div className='Payment'>
        <h2 className='Section-Header'>
          <FormattedMessage id='Billing.Payment.Header.Text' />
        </h2>
        <div data-tid='accepted-cards' className='Accepted-Cards'>
          <div aria-label='american-express' role='img' className='Amex'></div>
          <div aria-label='visa' role='img' className='Visa'></div>
          <div
            aria-label='master-card'
            role='img'
            className='Master-Card'
          ></div>
          <div
            // logo="discover"
            aria-label='discover'
            role='img'
            className='Discover'
          ></div>
          <div
            aria-label='diners-club'
            role='img'
            className='Diners-Club'
          ></div>
        </div>
        {/* TODO: the payment sections can be extracted into a PaymentSection component */}
        <div className='Payment-Section'>
          <label htmlFor='name-on-card' className='Payment-Form-Label'>
            <FormattedMessage id='Billing.Payment.Payment-Section.Name-On-Card.Label' />
          </label>
          <input
            type='text'
            onChange={handlePaymentChanges}
            className='nameOnCard'
            // onBlur={handleCardNumberBlur}
            value={cardInfo?.nameOnCard}
          />
        </div>
        <div className='Payment-Section'>
          <label htmlFor='card-number' className='Payment-Form-Label'>
            <FormattedMessage id='Billing.Payment.Payment-Section.Card-Number.Label' />
          </label>
          <input
            type='text'
            onChange={handlePaymentChanges}
            onBlur={handleCardNumberBlur}
            className='cardNumber'
            value={cardInfo?.cardNumber}
          />
          <div className='Payment-Validation-Error'>{cardInfo?.cardType}</div>
        </div>
        <div className='Payment-Section Expiration-Date-And-Security-Code'>
          <div className='Expiration-Date'>
            <label className='Payment-Form-Label'>
              <FormattedMessage id='Billing.Payment.Payment-Section.Expiration-Date.Label' />
            </label>
            <input
              type='month'
              onChange={handlePaymentChanges}
              onBlur={handleExpirationDateBlur}
              className='expirationDate'
              value={cardInfo?.expirationDate}
              maxLength={8}
            />
            <div>{cardInfo?.expirationDateValid}</div>
          </div>
          <div className='Security-Code'>
            <label className='Payment-Form-Label'>
              <FormattedMessage id='Billing.Payment.Payment-Section.Security-Code.Label' />
            </label>
            <input
              type='text'
              onChange={handlePaymentChanges}
              onBlur={handleSecurityCodeBlur}
              className='securityCode'
              value={
                cardInfo?.securityCode !== null ? cardInfo?.securityCode : ''
              }
              maxLength={3}
            />
            <div>{displayValidation('securityCode')}</div>
          </div>
        </div>
        <div></div>
      </div>
      <div className='Ticket-Insurance'>
        <h2 className='Section-Header'>
          <FormattedMessage id='Billing.Ticket-Insurance.Header.Text' />
        </h2>
        <p>
          <FormattedMessage id='Billing.Ticket-Insurance.Description.Text' />
        </p>
      </div>
    </div>
  )
}
