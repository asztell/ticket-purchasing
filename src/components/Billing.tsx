import { useContext, useCallback } from "react";
import { TicketPurchasingContext } from "../contexts";

export function Billing() {
  const { cardInfo, updateCardInfo } = useContext(TicketPurchasingContext);
  console.log(cardInfo);

  const handleCardNumberChanges = useCallback(
    (event: React.ChangeEvent<HTMLInputElement>) => {
      updateCardInfo({
        ...cardInfo,
        cardNumber: event.target.value,
      });
    },
    [cardInfo, updateCardInfo]
  );

  //   const handleCardNumberBlur = useCallback(() => {
  //     // TODO / MAYBE show credit card validation on blur
  //   }, [cardNumber, validateCardNumber]);

  const handleSecurityCodeChanges = useCallback(
    (event: React.ChangeEvent<HTMLInputElement>) => {
      updateCardInfo({
        ...cardInfo,
        securityCode: event.target.value,
      });
    },
    [cardInfo, updateCardInfo]
  );

  //   const handleSecurityCodeBlur = useCallback(() => {
  //     // TODO / MAYBE show security code validation on blur
  //   }, [securityCode, validateSecurityCode]);

  return (
    <div className="Billing">
      <h2>Billing</h2>
      <label
        htmlFor="card-number"
        style={{ marginRight: "10px", marginBottom: "20px" }}
      >
        Card Number
      </label>
      <input
        type="text"
        onChange={handleCardNumberChanges}
        // onBlur={handleCardNumberBlur}
        value={cardInfo.cardNumber}
        style={{ marginBottom: "10px" }}
      />
      <span>{cardInfo.cardType}</span>
      <br />
      <label style={{ marginRight: "10px" }}>Security Code</label>
      <input
        type="text"
        onChange={handleSecurityCodeChanges}
        // onBlur={handleSecurityCodeBlur}
        value={cardInfo.securityCode}
        maxLength={3}
        style={{ marginBottom: "20px" }}
      />
      <span>{cardInfo.securityCodeValid}</span>
    </div>
  );
}
