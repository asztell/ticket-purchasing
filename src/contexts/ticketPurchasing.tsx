import { createContext, useCallback, useMemo, useState } from "react";
import { Event } from "./events";

export type CardInfoType = {
  nameOnCard: string;
  cardNumber: string;
  cardType: string;
  securityCode: string;
  securityCodeValid: boolean;
  expirationDate: string;
  expirationDateValid: boolean;
};

export type TicketPurchasingContextType = {
  selectedEvent: Event | undefined;
  ticketsCounter: number;
  cardInfo: CardInfoType;
  updateSelectedEvent: (selectedEvent: Event) => void;
  updateTicketsCounter: (ticketsCounter: number) => void;
  updateCardInfo: (cardInfo: CardInfoType) => void;
};

export const TicketPurchasingContext =
  createContext<TicketPurchasingContextType>({} as TicketPurchasingContextType);

export function TicketPurchasingProvider({
  children,
}: {
  children: React.ReactNode;
}) {
  const [ticketsCounter, setTicketsCounter] = useState(0);
  const [selectedEvent, setSelectedEvent] = useState<Event | undefined>(
    undefined
  );
  const [cardInfo, setCardInfo] = useState({
    nameOnCard: "",
    cardNumber: "",
    cardType: "",
    securityCode: "",
    securityCodeValid: false,
    expirationDate: "",
    expirationDateValid: false,
  });

  const updateTicketsCounter = useCallback((ticketsCounter: number) => {
    setTicketsCounter(
      (prevTicketsCounter) => prevTicketsCounter + ticketsCounter
    );
  }, []);

  const updateSelectedEvent = useCallback((selectedEvent: Event) => {
    setSelectedEvent(selectedEvent);
  }, []);

  const updateCardInfo = useCallback((cardInfo: CardInfoType) => {
    const { expirationDate, cardNumber, securityCode } = cardInfo;
    const cardType = validateCardNumber(cardNumber);
    const securityCodeValid =
      securityCode.match(new RegExp("^[0-9]{3,4}$")) !== null;
    const expirationDateValid = validateExpirationDate(expirationDate);
    setCardInfo({
      ...cardInfo,
      cardType,
      securityCodeValid,
      expirationDateValid,
    });
  }, []);

  const value = useMemo(() => {
    return {
      selectedEvent,
      ticketsCounter,
      cardInfo,
      updateSelectedEvent,
      updateTicketsCounter,
      updateCardInfo,
    };
  }, [
    selectedEvent,
    ticketsCounter,
    cardInfo,
    updateSelectedEvent,
    updateTicketsCounter,
    updateCardInfo,
  ]);
  return (
    <TicketPurchasingContext.Provider value={value}>
      {children}
    </TicketPurchasingContext.Provider>
  );
}

export function validateCardNumber(cardNumber: string) {
  const visa = new RegExp("^4[0-9]{12}(?:[0-9]{3})?$");
  if (cardNumber.match(visa) !== null) return "Visa";
  const mastercard = new RegExp(
    "^5[1-5][0-9]{14}|^(222[1-9]|22[3-9]\\d|2[3-6]\\d{2}|27[0-1]\\d|2720)[0-9]{12}$"
  );
  if (cardNumber.match(mastercard) !== null) return "Mastercard";
  const amex = new RegExp("^3[47][0-9]{13}$");
  if (cardNumber.match(amex)) return "American Express";
  const discover = new RegExp("^6(?:011|5[0-9]{2})[0-9]{12}$");
  if (cardNumber.match(discover)) return "Discover";
  const dinersClub = new RegExp("^3(?:0[0-5]|[68][0-9])[0-9]{11}$");
  if (cardNumber.match(dinersClub)) return "Diners Club";
  if (cardNumber.length === 0) return "";
  return "Invalid";
}

export function validateExpirationDate(expirationDate: string) {
  const expirationDateValid =
    expirationDate.match(
      new RegExp("^(0[1-9]|1[0-2])\\/?([0-9]{4}|[0-9]{2})$")
    ) !== null && new Date(expirationDate) > new Date();
  return expirationDateValid;
}
