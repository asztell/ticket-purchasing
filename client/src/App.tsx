import type { ReactElement } from 'react'
import { Outlet } from 'react-router-dom'
import { IntlProvider } from 'react-intl'
import { Auth0Provider } from '@auth0/auth0-react'
import messages from './lang/en-US.json'
import { TicketPurchasingProvider, EventsProvider } from './contexts'
import { Header } from './components'
import './App.scss'

export function App(): ReactElement {
  const flattenedMessages = flattenMessages(messages)

  const auth0Domain = process.env.REACT_APP_AUTH0_DOMAIN as string
  const auth0ClientId = process.env.REACT_APP_AUTH0_CLIENT_ID as string
  const auth0RedirectUri = process.env.REACT_APP_AUTH0_REDIRECT_URI as string

  console.log('auth0Domain', auth0Domain)
  console.log('auth0ClientId', auth0ClientId)
  console.log('auth0RedirectUri', auth0RedirectUri)

  return (
    <IntlProvider locale={navigator.language} messages={flattenedMessages}>
      <EventsProvider>
        <TicketPurchasingProvider>
          <Auth0Provider
            domain={auth0Domain}
            clientId={auth0ClientId}
            authorizationParams={{
              // audience: process.env.REACT_APP_AUTH0_AUDIENCE,
              // scope: "openid profile email",
              redirect_uri: auth0RedirectUri
            }}
          >
            <Header />
            <div className='App'>
              <div className='App-Content'>
                <Outlet />
              </div>
            </div>
          </Auth0Provider>
        </TicketPurchasingProvider>
      </EventsProvider>
    </IntlProvider>
  )
}

export const flattenMessages = (
  nestedMessages: Record<string, any>,
  prefix = ''
): Record<string, string> => {
  if (nestedMessages === null) {
    return {}
  }
  return Object.keys(nestedMessages).reduce((messages, key) => {
    const value = nestedMessages[key]
    const prefixedKey = prefix !== '' ? `${prefix}.${key}` : key

    if (typeof value === 'string') {
      Object.assign(messages, { [prefixedKey]: value })
    } else {
      Object.assign(messages, flattenMessages(value, prefixedKey))
    }

    return messages
  }, {})
}
