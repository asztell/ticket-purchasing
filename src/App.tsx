import { Outlet } from "react-router-dom";
import { IntlProvider } from "react-intl";
// import { flatten } from "flat";
import messages from "./lang/en-US.json";
import { TicketPurchasingProvider } from "./contexts/ticketPurchasing";
import { EventsProvider } from "./contexts/events";
import "./App.scss";

export function App() {
  const flattenedMessages = flattenMessages(messages);
  console.log(flattenedMessages);
  return (
    <div className="App">
      <IntlProvider locale={navigator.language} messages={flattenedMessages}>
        <header className="App-Header"></header>
        <EventsProvider>
          <TicketPurchasingProvider>
            <div className="App-Content">
              <Outlet />
            </div>
          </TicketPurchasingProvider>
        </EventsProvider>
      </IntlProvider>
    </div>
  );
}

export const flattenMessages = (
  nestedMessages: Record<string, any>,
  prefix = ""
) => {
  if (nestedMessages === null) {
    return {};
  }
  return Object.keys(nestedMessages).reduce((messages, key) => {
    const value = nestedMessages[key];
    const prefixedKey = prefix ? `${prefix}.${key}` : key;

    if (typeof value === "string") {
      Object.assign(messages, { [prefixedKey]: value });
    } else {
      Object.assign(messages, flattenMessages(value, prefixedKey));
    }

    return messages;
  }, {});
};
