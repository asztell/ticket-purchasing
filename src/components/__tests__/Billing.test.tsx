import { render } from "@testing-library/react";
import { IntlProvider } from "react-intl";
import { flattenMessages } from "../../App";
import { Billing } from "../Billing";
import { TicketPurchasingProvider } from "../../contexts";

describe("<Billing />", () => {
  test("render Billing", () => {
    const { container } = render(
      <IntlProvider
        locale="en"
        messages={flattenMessages({
          // TODO: create a test helper version of the flattenMessages function
          // that adds Test to the beginning of each message
          Billing: {
            Delivery: {
              Header: {
                Text: "Test Delivery",
              },
              "Sub-Header": {
                Text: "Test Mobile - Free",
              },
              Description: {
                Text: "Test Your phone's your ticket. Locate your tickets in your account - or in your app. When you go mobile, your tickets will not be emailed to you or available for print.",
              },
            },
            Payment: {
              Header: {
                Text: "Test Payment",
              },
              "Payment-Section": {
                "Name-On-Card": {
                  Label: "Test Name on Card",
                },
                "Card-Number": {
                  Label: "Test Card Number",
                },
                "Expiration-Date": {
                  Label: "Test Expiration Date",
                },
                "Security-Code": {
                  Label: "Test Security Code",
                },
              },
            },
            "Ticket-Insurance": {
              Header: {
                Text: "Test Ticket Insurance",
              },
              Description: {
                Text: "Test Get reimbursed up to 100% with Event Ticket Insurance for only $28.00 per ticket ($56.00 total). *Offer not available after purchase is finalized.",
              },
            },
          },
        })}
      >
        <TicketPurchasingProvider>
          <Billing />
        </TicketPurchasingProvider>
      </IntlProvider>
    );
    expect(container).toMatchSnapshot();
  });
});
