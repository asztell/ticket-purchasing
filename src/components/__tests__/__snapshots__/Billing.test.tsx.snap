// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Billing /> render Billing 1`] = `
<div>
  <div>
    <div
      class="Delivery"
    >
      <h2
        class="Section-Header"
      >
        Test Delivery
      </h2>
      <h3>
        Test Mobile - Free
      </h3>
      <p>
        Test Your phone's your ticket. Locate your tickets in your account - or in your app. When you go mobile, your tickets will not be emailed to you or available for print.
      </p>
    </div>
    <div
      class="Payment"
    >
      <h2
        class="Section-Header"
      >
        Test Payment
      </h2>
      <div
        class="Accepted-Cards"
        data-tid="accepted-cards"
      >
        <div
          aria-label="american-express"
          class="Amex"
          role="img"
        />
        <div
          aria-label="visa"
          class="Visa"
          role="img"
        />
        <div
          aria-label="master-card"
          class="Master-Card"
          role="img"
        />
        <div
          aria-label="discover"
          class="Discover"
          role="img"
        />
        <div
          aria-label="diners-club"
          class="Diners-Club"
          role="img"
        />
      </div>
      <div
        class="Payment-Section"
      >
        <label
          class="Payment-Form-Label"
          for="name-on-card"
        >
          Test Name on Card
        </label>
        <input
          type="text"
          value=""
        />
      </div>
      <div
        class="Payment-Section"
      >
        <label
          class="Payment-Form-Label"
          for="card-number"
        >
          Test Card Number
        </label>
        <input
          type="text"
          value=""
        />
        <span
          class="Payment-Validation-Error"
        />
      </div>
      <div
        class="Payment-Section"
      >
        <label
          class="Payment-Form-Label"
        >
          Test Expiration Date
        </label>
        <input
          maxlength="8"
          type="month"
          value=""
        />
        <span />
      </div>
      <div
        class="Payment-Section"
      >
        <label
          class="Payment-Form-Label"
        >
          Test Security Code
        </label>
        <input
          maxlength="3"
          type="text"
          value=""
        />
        <span />
      </div>
      <div />
    </div>
    <div
      class="Ticket-Insurance"
    >
      <h2
        class="Section-Header"
      >
        Test Ticket Insurance
      </h2>
      <p>
        Test Get reimbursed up to 100% with Event Ticket Insurance for only $28.00 per ticket ($56.00 total). *Offer not available after purchase is finalized.
      </p>
    </div>
  </div>
</div>
`;
